base/abstractarraymath.jl

```diff
@@ -91,6 +91,10 @@ imag{T<:Real}(x::AbstractArray{T}) = zero(x)
 
 # index A[:,:,...,i,:,:,...] where "i" is in dimension "d"
 
+_slice_inds(new_inds::Tuple, inds::Tuple, d, i) =
+    _slice_inds((new_inds..., length(new_inds) + 1 == d ? i : first(inds)), tail(inds), d, i)
```

Comment:
I'd consider `ifelse` here since otherwise you'll get one branch per dimension. Might also consider `@inline`ing this.